File: fleet/fleet_manager/report/vehicle_income_due/vehicle_income_due.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Vehicle Income Due"] = {
//    "onload":function(frm){
//        frappe.msgprint(__("jjjj"))
//    },
	"filters": [
		{
			"fieldname":"vehicle",
			"label": __("Vehicle"),
			"fieldtype": "Link",
			"options": "Vehicle",
			"width": "80"
		},
		{
        	"fieldname":"owner",
        	"label": __("Owner"),
        	"fieldtype": "Link",
        	"options": "Customer"
        }

	],
	"formatter":function (row, cell, value, columnDef, dataContext, default_formatter) {
            value = default_formatter(row, cell, value, columnDef, dataContext);
            var date=new Date(dataContext.next_payment_date);
//            var parts =dataContext.due_on.split('-');
//            var mydate = new Date(parts[0], parts[1] - 1, parts[2]);
//            frappe.msgprint(__(mydate))
//            var date=new Date(dataContext.due_on);
//            var today=new Date()
//              var tom= frappe.datetime.add_days(frappe.datetime.get_today(), 7)
//              var diff=frappe.datetime.get_day_diff(tom, frappe.datetime.get_today())
//              frappe.msgprint(__(frappe.datetime.get_day_diff(tom, frappe.datetime.get_today())))
//              frappe.msgprint(__(frappe.datetime.get_day_diff(dataContext.next_payment_date, frappe.datetime.get_today())))


            var date1 = new Date(frappe.datetime.get_today());
            var date2 = new Date(dataContext.next_payment_date);
            var diffDays = parseInt((date2 - date1) / (1000 * 60 * 60 * 24));

            if (columnDef.id == "Amount" || columnDef.id != "Amount")
                {
                    if(diffDays<0){
                        value = "<span style='color:red!important;font-weight:bold;'>" + value + "</span>";
                    }
                    else if(diffDays==0){
                        value = "<span style='color:green!important;font-weight:bold;'>" + value + "</span>";

                    }


                }
            return value;
        }
};

File: fleet/fleet_manager/report/vehicle_income_due/vehicle_income_due.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import msgprint, _

def execute(filters=None):
	if not filters: filters = {}

	advances_list = get_advances(filters)
	columns = get_columns()

	if not advances_list:
		msgprint(_("No record found"))
		return columns, advances_list

	data = []
	for advance in advances_list:
		row = [advance.name, advance.vehicle,advance.owner, advance.date, advance.next_payment_date, advance.amount]
		data.append(row)

	return columns, data


def get_columns():
	return [
		{
			"label": _("Dues No"),
			"fieldname": "due_no",
			"fieldtype": "Link",
			"options": "Vehicle Income",
			"width": 120
		},
		{
			"label": _("Vehicle"),
			"fieldname": "vehicle",
			"fieldtype": "Link",
			"options": "Vehicle",
			"width": 120
		},
		{
			"label": _("Owner"),
			"fieldname": "owner",
			"fieldtype": "Link",
			"options": "Customer",
			"width": 120
		},
		{
			"label": _("Lat Payment Date"),
			"fieldname": "last_payment_date",
			"fieldtype": "Date",
			"width": 120
		},
		{
			"label": _("Due On"),
			"fieldname": "next_payment_date",
			"fieldtype": "Date",
			"width": 120
		},
		{
			"label": _("Amount"),
			"fieldname": "amount",
			"fieldtype": "Currency",
			"width": 120
		}
	]

def get_conditions(filters):
	conditions = ""

	if filters.get("vehicle"):
		conditions += " and vehicle = %(vehicle)s"
	if filters.get("owner"):
		conditions += " and owner = %(owner)s"
	# if filters.get("next_payment_date"):
	# 	conditions += " and date<=%(next_payment_date)s"

	return conditions

def get_advances(filters):
	conditions = get_conditions(filters)
	return frappe.db.sql("""select name, vehicle, owner, date, next_payment_date, 
	    DATEDIFF(next_payment_date, date ) *1000 as amount from `tabVehicle Income`
		where age='Last' %s order by date ASC""" %
						 conditions, filters, as_dict=1)